components/motor/dimensionengineering/sabertooth.go

```diff
@@ -0,0 +1,412 @@
+// Package dimensionengineering contains implementations of the dimensionengineering motor controls
+package dimensionengineering
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"sync"
+
+	"github.com/edaniels/golog"
+	"github.com/jacobsa/go-serial/serial"
+	"github.com/mitchellh/mapstructure"
+	"go.viam.com/utils/usb"
+
+	"go.viam.com/rdk/components/motor"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/operation"
+	"go.viam.com/rdk/registry"
+)
+
+const (
+	// https://www.dimensionengineering.com/datasheets/Sabertooth2x60.pdf
+	modelName = "sabertooth"
+)
+
+// controllers is global to all instances, mapped by serial device.
+var (
+	globalMu    sync.Mutex
+	controllers map[string]*controller
+	usbFilter   = usb.SearchFilter{}
+)
+
+// controller is common across all Sabertooth motor instances sharing a controller.
+type controller struct {
+	mu           sync.Mutex
+	port         io.ReadWriteCloser
+	serialDevice string
+	logger       golog.Logger
+	activeAxes   map[int]bool
+	testChan     chan []byte
+	address      int // 128-135
+}
+
+// Motor is a single axis/motor/component instance.
+type Motor struct {
+	c       *controller
+	Channel int
+	jogging bool
+	opMgr   operation.SingleOperationManager
+}
+
+// GoFor Not supported.
+func (m *Motor) GoFor(ctx context.Context, rpm, revolutions float64, extra map[string]interface{}) error {
+	return errors.New("not supported")
+}
+
+// IsPowered returns if the motor is currently on or off.
+func (m *Motor) IsPowered(ctx context.Context, extra map[string]interface{}) (bool, error) {
+	return m.jogging, nil
+}
+
+// Config adds DimensionEngineering-specific config options.
+type Config struct {
```

Comment:
[nit] add `...,omitempty"` for unneeded fields like here:
https://github.com/viamrobotics/rdk/blob/0e659a0723aeb3d8b0b3038a2e7b1735732305ba/components/gantry/oneaxis/oneaxis.go#L35

and check them with a `Validate` function like here: 
https://github.com/viamrobotics/rdk/blob/0e659a0723aeb3d8b0b3038a2e7b1735732305ba/components/movementsensor/imuvectornav/imu.go#L37